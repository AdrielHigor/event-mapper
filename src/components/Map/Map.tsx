import React from "react";
import "./Map.css";
import GoogleMapReact from "google-map-react";
import Marker from "../Marker/Marker";
import constants from "../../utils/Constants";
import { IEventResponse } from "../../utils/interfaces/base";

interface IMap {
  location: GoogleMapReact.Coords;
  zoomLevel: number;
  handleMapClick?: (event: GoogleMapReact.ClickEventValue) => void;
  markers?: Array<IEventResponse>;
}

const Map = ({ location, zoomLevel, handleMapClick, markers }: IMap) => (
  <div className="map">
    <div className="google-map">
      <GoogleMapReact
        bootstrapURLKeys={{ key: constants.google_maps_api_key }}
        defaultCenter={location}
        defaultZoom={zoomLevel}
        yesIWantToUseGoogleMapApiInternals
        onClick={handleMapClick}
      >
        {markers?.map((marker) => (
          <Marker
            key={marker.id}
            lat={marker.location.point.coordinates[1]}
            lng={marker.location.point.coordinates[0]}
          />
        ))}
      </GoogleMapReact>
    </div>
  </div>
);

export default Map;
