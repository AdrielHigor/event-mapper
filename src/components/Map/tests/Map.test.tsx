import renderer from "react-test-renderer";
import Map, { IMap } from "../Map";

jest.mock("../Map.css", () => ({}));

describe("Marker", () => {
  let component: renderer.ReactTestRenderer;

  const mockMarkers = [
    {
      id: 1,
      name: "Test event",
      description: "Test description",
      startDateTime: "2023-05-01T12:00:00.000Z",
      endDateTime: "2023-05-01T14:00:00.000Z",
      createdAt: "2023-05-01T12:00:00.000Z",
      deletedAt: "2023-05-01T12:00:00.000Z",
      updatedAt: "2023-05-01T12:00:00.000Z",
      location: {
        id: 1,
        point: {
          coordinates: [0, 1],
          crs: {
            properties: {
              type: "test",
            },
            type: "test",
          },
          type: "test",
        },
      },
    },
    {
      id: 2,
      name: "Test event 2",
      description: "Test description 2",
      startDateTime: "2023-05-01T12:00:00.000Z",
      endDateTime: "2023-05-01T14:00:00.000Z",
      createdAt: "2023-05-01T12:00:00.000Z",
      deletedAt: "2023-05-01T12:00:00.000Z",
      updatedAt: "2023-05-01T12:00:00.000Z",
      location: {
        id: 2,
        point: {
          coordinates: [1, 2],
          crs: {
            properties: {
              type: "test",
            },
            type: "test",
          },
          type: "test",
        },
      },
    },
  ];

  const defaultProps: IMap = {
    location: { lat: 0, lng: 0 },
    zoomLevel: 15,
    markers: mockMarkers,
    handleMarkerClick: jest.fn(),
    handleMapClick: jest.fn(),
  };

  beforeEach(() => {
    component = renderer.create(<Map {...defaultProps} />);
  });

  it("should render correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should call onClick function when marker clicked", () => {
    const markers = component.root.findAllByProps({
      className: "marker",
    });

    markers[0].props.onClick();
    expect(defaultProps.handleMarkerClick).toHaveBeenCalledTimes(1);
    markers[1].props.onClick();
    expect(defaultProps.handleMarkerClick).toHaveBeenCalledTimes(2);
  });
});
