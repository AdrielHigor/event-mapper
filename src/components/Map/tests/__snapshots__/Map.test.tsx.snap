// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker should render correctly 1`] = `
<div
  className="map"
>
  <div
    className="google-map"
  >
    <div>
      <img
        alt="Location Marker"
        className="marker"
        onClick={[Function]}
        src="mockImage"
      />
      <img
        alt="Location Marker"
        className="marker"
        onClick={[Function]}
        src="mockImage"
      />
    </div>
  </div>
</div>
`;
